routing/rt/graph/edgepath.go

```diff
@@ -0,0 +1,139 @@
+package graph
+
+import (
+	"container/list"
+	"fmt"
+)
+
+// Path in a graph represented as a list of edges.
+// It is useful because it contains information
+// about each edge (like EdgeID)
+type EdgePath struct {
+	Source ID
+	Target ID
+	Edges  []Edge
+}
+
+func (ep *EdgePath) String() string {
+	return fmt.Sprintf("EdgePath[%v %v %v]", ep.Source, ep.Target, ep.Edges)
+}
+
+// Adds to EdgePath. Return error if second source is different from first target
+// It does not deeply copy edges. So modifying input path may modify result
+func (ep1 *EdgePath) Add (ep2 *EdgePath) (*EdgePath, error) {
```

Comment:
It uses EdgePath in an immutable manner. It simplifies concurrent usage of EdgePath.